model/__tests__/lollipop_keys.test.ts

```diff
@@ -0,0 +1,162 @@
+import * as E from "fp-ts/lib/Either";
+import * as tk from "timekeeper";
+
+import { Container } from "@azure/cosmos";
+import { AssertionRef } from "@pagopa/io-functions-commons/dist/generated/definitions/lollipop/AssertionRef";
+import { AssertionTypeEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/lollipop/AssertionType";
+import { CosmosResource } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import {
+  AssertionFileName,
+  LolliPOPKeysModel,
+  PendingPopDocument,
+  PopDocument,
+  PopDocumentStatusEnum,
+  RetrievedPopDocument,
+  TTL_VALUE_AFTER_UPDATE,
+  TTL_VALUE_FOR_RESERVATION
+} from "../lollipop_keys";
+
+const aFiscalCode = "RLDBSV36A78Y792X" as FiscalCode;
+const anAssertionRef = "sha256-p1NY7sl1d4lGvcTyYS535aZR_iJCleEIHFRE2lCHt-c" as AssertionRef;
+const anAssertionFileName = `${aFiscalCode}-${anAssertionRef}` as AssertionFileName;
+const aPubKey = "aValidPubKey" as NonEmptyString;
+
+const aCosmosResourceMetadata: Omit<CosmosResource, "id"> = {
+  _etag: "_etag",
+  _rid: "_rid",
+  _self: "_self",
+  _ts: 1
+};
+
+const aPendingPopDocument: PendingPopDocument = {
+  assertionRef: anAssertionRef,
+  pubKey: aPubKey,
+  status: PopDocumentStatusEnum.PENDING
+};
+
+const aPopDocument: PopDocument = {
+  assertionFileName: anAssertionFileName,
+  assertionRef: anAssertionRef,
+  assertionType: AssertionTypeEnum.OIDC,
+  expiredAt: new Date(),
+  fiscalCode: aFiscalCode,
+  pubKey: aPubKey,
+  status: PopDocumentStatusEnum.VALID
+};
+
+const aRetrievedPopDocument: RetrievedPopDocument = {
+  id: `${aPopDocument.assertionRef}-${"0".repeat(16)}` as NonEmptyString,
+  ...aCosmosResourceMetadata,
+  ...aPopDocument,
+  ttl: TTL_VALUE_AFTER_UPDATE, // 2y
+  version: 0 as NonNegativeInteger
+};
+
+const mockCreateItem = jest.fn();
+const mockUpsert = jest.fn();
+const mockFetchAll = jest.fn();
+
+mockFetchAll.mockImplementation(async () => ({
+  resources: []
+}));
+
+const containerMock = ({
+  items: {
+    create: mockCreateItem,
+    query: jest.fn(() => ({
+      fetchAll: mockFetchAll
+    })),
+    upsert: mockUpsert
+  }
+} as unknown) as Container;
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const mockedNowTime = new Date(TTL_VALUE_FOR_RESERVATION * 1000);
+
+beforeAll(() => {
+  tk.freeze(mockedNowTime);
```

Comment:
Why are we using a third party library here?

Reply:
Jest has a native method but we should update jest to v27